wlroots/render/renderer.scm

```diff
@@ -36,6 +38,17 @@
   (ffi:int "wlr_renderer_clear" `(* *))
   (% (unwrap-wlr-renderer renderer) (color->pointer color)))
 
+(define-wlr-procedure (wlr-render-texture-with-matrix r texture matrix alpha)
+  (ffi:int8 "wlr_render_texture_with_matrix" (list '* '* '* ffi:float))
+  (not (zero? (% (unwrap-wlr-renderer r)
+                 (unwrap-wlr-texture texture)
+                 (if (>= (length matrix) 9)
+                     (bytestructure->pointer
+                       (bytestructure (bs:vector (length matrix) float)
+                                      (list->vector matrix)))
+                   (error "warning: list argument is too small; matrix requires, at least, 9"))
```

Comment:
don't use `warning: ` begin, and use `srfi-34` `raise` like
`(raise (condition
                (&message
                 (message  "Not a 9 length vector or list"))))`